src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -615,5 +619,51 @@ private static Assembly SgenAssemblyResolver(object source, ResolveEventArgs e)
 
             return null;
         }
+
+        private string[] ParseResponseFile(string[] args)
+        {
+            var parsedArgs = new List<string>();
+            foreach (string arg in args)
+            {
+                if (!arg.StartsWith("@"))
+                {
+                    parsedArgs.Add(arg);
+                }
+                else
+                {
+                    string responseFileName = arg.Substring(1);
+                    foreach (string line in ReadResponseFile(responseFileName))
+                    { 
+                        int i = line.Trim().IndexOf(' ');
+                        if (i < 0)
+                        {
+                            parsedArgs.Add(line);
+                        }
+                        else
+                        {
+                            parsedArgs.Add(line.Substring(0, i));
+                            parsedArgs.Add(line.Substring(i + 1));
+                        }
+
+                    }
+                }
+            }
+            return parsedArgs.ToArray();
+        }
+
+        private string[] ReadResponseFile(string responseFileName)
+        {
+            string[] lines;
+            try
+            {
+                lines = File.ReadAllLines(responseFileName);
+            }
+            catch (Exception e)
+            {
+                WriteError(e, true);
+                throw;
```

Comment:
We'd better not break the sgen if there's any error thrown when read the file.

Reply:
@huanwu When exceptions are thrown during reading rsp file, do we still want sgen to proceed?